mm2src/coins/utxo/rpc_clients/electrum_rpc/client.rs

```diff
@@ -0,0 +1,1056 @@
+        to_height: u64,
+    ) -> UtxoRpcFut<(HashMap<u64, BlockHeader>, Vec<BlockHeader>)> {
+        let coin_name = self.coin_ticker.clone();
+        if from_height == 0 || to_height < from_height {
+            return Box::new(futures01::future::err(
+                UtxoRpcError::Internal("Invalid values for from/to parameters".to_string()).into(),
+            ));
+        }
+        let count: NonZeroU64 = match (to_height - from_height + 1).try_into() {
+            Ok(c) => c,
+            Err(e) => return Box::new(futures01::future::err(UtxoRpcError::Internal(e.to_string()).into())),
+        };
+        Box::new(
+            self.get_block_headers_from(server_address, from_height, count)
+                .map_to_mm_fut(UtxoRpcError::from)
+                .and_then(move |headers| {
+                    let (block_registry, block_headers) = {
+                        if headers.count == 0 {
+                            return MmError::err(UtxoRpcError::Internal("No headers available".to_string()));
+                        }
+                        let len = CompactInteger::from(headers.count);
+                        let mut serialized = serialize(&len).take();
+                        serialized.extend(headers.hex.0.into_iter());
+                        drop_mutability!(serialized);
+                        let mut reader =
+                            Reader::new_with_coin_variant(serialized.as_slice(), coin_name.as_str().into());
+                        let maybe_block_headers = reader.read_list::<BlockHeader>();
+                        let block_headers = match maybe_block_headers {
+                            Ok(headers) => headers,
+                            Err(e) => return MmError::err(UtxoRpcError::InvalidResponse(format!("{:?}", e))),
+                        };
+                        let mut block_registry: HashMap<u64, BlockHeader> = HashMap::new();
+                        let mut starting_height = from_height;
+                        for block_header in &block_headers {
+                            block_registry.insert(starting_height, block_header.clone());
+                            starting_height += 1;
+                        }
+                        (block_registry, block_headers)
+                    };
+                    Ok((block_registry, block_headers))
+                }),
+        )
+    }
+
+    pub(crate) fn get_servers_with_latest_block_count(&self) -> UtxoRpcFut<(Vec<String>, u64)> {
+        let selfi = self.clone();
+        let fut = async move {
+            let addresses = selfi.connection_manager.get_all_server_addresses();
+            let futures = addresses
+                .into_iter()
+                .map(|address| {
+                    selfi
+                        .get_block_count_from(&address)
+                        .map(|response| (address, response))
+                        .compat()
+                })
+                .collect::<Vec<_>>();
```

Comment:
good catch, this is not respecting the `max_connected` thresh, as these connections will be open in the same time (till they timeout and close in their own). we should disconnect after querying the connection (or let the manager disconnect it as `not needed`, but only if it's not maintained).

> shouldn't we use active addresses

I guess for this method since we want the latest block servers, we should test all of them.

Reply:
> Let's add this to the tracking issue to handle later.

I can't see this added here https://github.com/KomodoPlatform/komodo-defi-framework/issues/2110#issue-2281124916